### PR TITLE
host: fix last-panic/boot-fail commands

### PR DESCRIPTION
**Pull request by cbiffle**
Memory layout changed in Hubris.

### CODE REVIEW

**Comment by hawkw** [comment]
rustfmt looks a little bit mad: https://github.com/oxidecomputer/humility/actions/runs/9357979583/job/25758919977?pr=492

File: cmd/host/src/lib.rs [hunk 1]
```diff
@@ -77,35 +78,114 @@ struct HostArgs {
...
+            }
+            b'\n' | b'\r' | b'\t' | 0x20..=0x7E => {
+                // Pass through basic text characters that we expect we might
+                // see in a message.
+                buf.push(b as char);
+            }
+            _ => {
+                // Escape any other non-printable characters.
+                buf.push_str("\\x");
+                buf.push_str(&format!("{b:02X}"));
```
Comment (faithanalog) [inline cmd/host/src/lib.rs:163]: yay zero-padding! i have a newfound appreciation for it

File: cmd/host/src/lib.rs [hunk 2]
```diff
@@ -77,35 +78,114 @@ struct HostArgs {
     cmd: HostCommand,
 }
 
-fn read_var(
+static SEPARATE_HOST_BOOT_FAIL_NAME: &str = "LAST_HOST_BOOT_FAIL";
+
+static SEPARATE_LAST_HOST_PANIC_NAME: &str = "LAST_HOST_PANIC";
+
+static HOST_STATE_BUF_NAME: &str =
+    "task_host_sp_comms::ServerImpl::claim_static_resources::BUFS";
```
Comment (faithanalog) [inline cmd/host/src/lib.rs:86]: i hope the buffer this is referencing never changes names

**Review by faithanalog — APPROVED** [review]

File: cmd/host/src/lib.rs (hunk 2 above)
Comment (cbiffle) [reply cmd/host/src/lib.rs:86]: Separately, I'm preparing a commit in Hubris that adds a big freakin' comment to that buffer saying something like

```rust
// HEY
//
// HEY LISTEN
//
// HUMILITY DEPENDS ON THE NAME OF THIS VARIABLE
```

**Comment by cbiffle** [comment]
> rustfmt looks a little bit mad: https://github.com/oxidecomputer/humility/actions/runs/9357979583/job/25758919977?pr=492

sooooo aaaaaangry (fixed)

**Comment by cbiffle** [comment]
@faithanalog I'm gradually remembering how to write non-awful code in Humility, and I've managed to simplify the implementation a fair amount by leaning more heavily on the `reflect` module (which, ironically, I wrote).

PTAL.

**Pull request merged**